src/js/components/DataTable/Header.js

```diff
@@ -55,6 +55,9 @@ const buttonStyle = ({ theme }) => {
   const [layoutProps, , iconProps] = separateThemeProps(theme);
 
   if (layoutProps) {
+    // if cell is sortable, we want pad to be applied
+    // to the button instead of the cell
+    layoutProps.pad = theme.dataTable.header.pad;
```

Comment:
I think we want to pass `cellProps.pad` to StyledHeaderCellButton. It is already accounting for styling coming from the props and the theme and falling back to theme.table styling as needed.